docs/scos/dev/guidelines/nfr-guidelines.md

```diff
@@ -0,0 +1,87 @@
+to your project. Not all of these NFRs may be applicable in every case, so you SHOULD use this list as a guide or template to help to start
+with your own NFRs. Be sure to tailor the list to the specific requirements and constraints of your project, and to prioritize and organize
+the NFRs in a logical manner.
+
+{% endinfo_block %}
+
+### Availability
+Software architecture & design must ensure that there is no negative impact on application availability.
+
+Example NFRs:
+* Avoid designing the application to limit its own start-up or shut-down.
+  * Avoid design patterns or coding practices that cause the application to exit unexpectedly in the middle of execution.
+  * Avoid using flags or other mechanisms to block the application from starting.
+  * Avoid setting limits on the number of concurrent processes that can be run by the application.
+* Avoid designing the application to lock shared resources at the application level.
+  * For example, if the application uses a database, it should not lock the database tables or rows that it reads or writes to, as this can cause other parts of the system to become unavailable.
+* Align resource consumption with expected workloads.
+  * For example, if using PHP-FPM, ensure that the number of worker processes is sufficient to handle the expected workload without overloading the system.
+
+### Security
+* Make sure to define and follow your own team's [Security best practices](/docs/scos/dev/guidelines/security-guidelines.html).
+
+### Deployability
+* The same release candidate / branch MUST be re-deployable without side-effects.
+* The deploy scripts MUST NOT break the behaviour of the current system.
+  * Deploy script elements MUST be configured according to your project setup following the [Deployment Pipeline](/docs/cloud/dev/spryker-cloud-commerce-os/configure-deployment-pipelines/deployment-pipelines.html) process.
+  * The to-be deployed (version N+1) application MUST upgrade the data structures and constants and data sets without causing downtime and without failing or losing/compromising any (version N+1 or N) functionality.
+  * The to-be deployed (version N+1) message consumers MUST be BC with the present (<N+1) messages and broker structure.
+  * The deployed (version N+1) application CAN clean up obsolete data structures # constants and data sets if it causes no downtime, unless it's breaking version N.
+  
+### Rollback-ability
+The rollback scripts MUST NOT break the behaviour of current system.
+
+* Rollback script elements MUST be configured according to your project setup following the Rollback Pipeline process.
+* The deployed (version N+1) application can rollback current version (N+1) data structures # constants and data sets to previous version (N) without causing downtime.
+
+### Deployability, Rollback-ability, Consistency, Monitor-ability
```

Comment:
This section is something that we could remove.
The asynchronous scripts are really unique to Aldi. Although you can not put together a decent SCCOS without it, our boilerplates and deployment pipelines don't offer this ability; and if you design these on project level, their NFRs might be far away from these.